enctests/testframework/test_suite.py

```diff
@@ -0,0 +1,191 @@
+import os
+import sys
+import time
+import shlex
+import yaml
+import pathlib
+from copy import deepcopy
+from typing import Any
+
+try:
+    from yaml import CSafeLoader as SafeLoader
+    from yaml import CSafeDumper as SafeDumper
+
+except ImportError:
+    from yaml import SafeLoader, SafeDumper
+
+"""
+A test of classes to help manage loading and querying the test suites and the associated source content.
+"""
+
+# Test config files
+from .utils import (
+    create_clip,
+    get_media_info,
+    get_source_path,
+    get_nearest_model,
+    get_test_metadata_dict,
+    get_source_metadata_dict
+)
+
+
+class BaseYamlConfig:
+    """
+    Base class for reading the yaml file in, and storing the location of the file.
+    """
+    def __init__(self,
+                 test_config_file: pathlib.Path):
+        self.__config__ = {}
+        self.__config_path = None
+        self.parse_config_file(test_config_file)
+
+    def dictcopy(self):
+        return deepcopy(self.__config__)
+
+    def parse_config_file(self, path: pathlib.Path):
+        self.__config_path = path
+        path = path.absolute().as_posix()
+        self.__config__ = {}
+        with open(path, 'rt') as f:
+            self.__config__ = list(yaml.load_all(f, SafeLoader))
+        return self.__config__
```

Comment:
We are sadly not being consistent with our yml files, for the main test configs, we are expecting a list, but for the individual sequence config files, we are expecting a dictionary. 

Part of me does feel like we perhaps should be a bit more rigorous with our test-config files, and have a "tests" section, and a "reports" section, so that way its explicitly a dictionary.

Reply:
Yeah, I was afraid we were inconsistent from way back. Let's resolve this in a separate PR as suggested in #72